### ISSUE TITLE
Cannot switch to input source "wifi" on Soundbar Q80R

### ISSUE THREAD

**Issue by buentead** [PiotrMachowski/Home-Assistant-custom-components-SmartThings-Soundbar#25]
Using the service `media_player.select_source` I can switch the input source of my Samsung Q80R soundbar to `hdmi1` and `hdmi2`. But I'm unable to switch to `wifi`. Also I'm unable to switch to `digital`.

_Note:_ I'm able to switch to any input source using the Samsung SmartThings App on my mobile phone.

Thank you for your work done!

**Comment by buentead** [comment]
I just realized, that I cannot switch to any source anymore.

**Comment by PiotrMachowski** [comment]
Are there any errors in logs?

**Comment by buentead** [comment]
No, unfortunately not. I checked in /config/logs and on hassio/system.

**Comment by buentead** [comment]
I run the API by POSTMAN myself:
```
HTTP POST https://api.smartthings.com/v1/devices/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX/commands
{
    'commands': [
        {
            'component': 'main',
            'capability': 'mediaInputSource',
            'command': 'setInputSource',
            'arguments': [
                'wifi'
            ]
        }
    ]
}
```
Which returns:
```
HTTP status: 200 OK
{
    "results": [
        {
            "id": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
            "status": "ACCEPTED"
        }
    ]
}
```
Now, I assume it is a problem on the Soundbar Q80R itself and not in your component.

Do you know what the problem may be?

**Comment by PiotrMachowski** [comment]
Sometimes they act wonky, you can try resetting it and it might be ok after adding it to SmartThings again

**Comment by PiotrMachowski** [comment]
Sometimes they act wonky, you can try resetting it and it might be ok after adding it to SmartThings again

**Comment by haakonhh** [comment]
I have exactly the same problem on my HW-Q70R (HW-Q76R). I belive this worked a while ago, but the last weeks/month i can no longer change input to wifi from this component. I have tried to factory reset the soundbar, but the issue still persists. Change of input from the SmartThings app works, so that makes it even more strange.

**Comment by salvq** [comment]
@buentead thank you for the guide however I was not able to set the media renderer in point 3, can you help to clarify.

How did you set the samsung as DLNA Renderer ? I have missing soundbar in the list in HA however my samsung is visible in HA, in network and also in smartthings and have working integration from PiotrMachowski

![image](https://user-images.githubusercontent.com/43242348/167311567-30c5adf7-9f3b-4e94-8825-d9c8e05b6ff1.png)

**Comment by buentead** [comment]
You need to add the [DLNA Digital Media Renderer](https://www.home-assistant.io/integrations/dlna_dmr). Then, the Soundbar will be discovered automatically.

**Comment by salvq** [comment]
> You need to add the [DLNA Digital Media Renderer](https://www.home-assistant.io/integrations/dlna_dmr). Then, the Soundbar will be discovered automatically.

Actualy this is what I did and got the output above...samsung soundbar did not showed up in the list of supported devices. Have you enabled the dlna for samsung somewhere ?

**Comment by buentead** [comment]
No. Not that I'm aware off. Could it be, that your device is `ignored` or `disabled` from the HA Integration?
![image](https://user-images.githubusercontent.com/66123585/167663022-0d8ab567-6d5d-4dd7-b5aa-fdb5aef80f60.png)

In my situation, the soundbar just showed up; I got a message, that a new device was discovered and it was listed there.

Is your soundbar in the same LAN subnet as HA so it can be 'seen'?

**Comment by salvq** [comment]
Using uPnP devices showed up interesting thing, looks like HW-Q900a does not support DLNA which is very suprising considering that this is one of the Samsung high end subwoofers :(

Device is specified like dialreceiver not MediaRenderer, I am waiting for feedback from support to clarify...

![image](https://user-images.githubusercontent.com/43242348/168440630-0147364d-6a90-4430-afae-62ea4c55b54b.png)

**Comment by buentead** [comment]
Thanks for the feedback. I never saw 'dialreceiver' as a device in my environment. Is this a device using the [Discovery and Launch (DIAL) protocol developed by Netflix and YouTube with the help from Sony and Samsung](https://en.wikipedia.org/wiki/Discovery_and_Launch)? Will be interesting to hear from Samsung support.

**Comment by salvq** [comment]
I got a feedback that Q900a does not support DLNA. However I made a request to the development team to make Q900a DLNA capable but doubt that this will happen
